src/Registration/RegistrationService.php

```diff
@@ -127,4 +127,24 @@ public function registerConfirm(RequestInterface $request): ResponseInterface
 
         return (new Psr17Factory())->createResponse(204);
     }
+
+    private function sanitizeShopUrl(string $shopUrl): string
+    {
+        $parsedUrl = parse_url($shopUrl);
+
+        $protocol = $parsedUrl['scheme'] ?? '';
+        $host = $parsedUrl['host'] ?? '';
+        $path = $parsedUrl['path'] ?? '';
```

Comment:
port is missing or is it part of host?

Reply:
You are right, the port was missing and I added it 👍